Add 3D PCA Plot

catterbu opened the pull request:
This adds a 3D PCA plot called `pca_plot3d` with a wrapper on `scatterplot3d`. Also, the title for both PCA plots refer to the amount of variance explained.

nsh87 committed:
Add helper function to optimize num_cols in cluster_boxplot()
  - Reduces the number of columns of plots as the number of clusters
  increases so that the boxplots and x-axis labels don't get too close

nsh87 commented:
@catterbu, this needs to be rebased off of my dev before I can merge, in effect removing commit 2eeefb0 (`Determine k_best using NbClust`). I can't merge that commit yet, it should only be in PR #28.

nsh87 commented:
Sorry, not in effect removing that commit, what am I saying. You have to remove the commit by hand from this branch, rebasing won't do the trick.

nsh87 commented:
Nice job on this. I wouldn't focus on this comment now, but in 3D it gets hard to interpret a static image with more than a few points:

![screen shot 2015-07-20 at 6 35 10 pm](https://cloud.githubusercontent.com/assets/4327717/8791498/7c38e56c-2f10-11e5-90e0-838be4b431d0.png)

I think we might want to consider holding further development on any _static_ 3D visualizations and aim to do all 3D stuff in JS. If you can get the coordinates of each of the points in the PCA we can build something like this very quickly: https://jsfiddle.net/rd26u1zj/1/. Let me know if possible.

nsh87 committed:
Merge pull request #35 from nsh87/automate_boxplot_fontsize

Reduce overlap of clust_boxplot axis labels and tightness of graphs at high num_clust

catterbu committed:
Add pca_plot3d to cluster_plotting
  - Add cluster_plot3d function
  - Add documentation
  - Add scatterplot3d to imports
  - Modify other docs to include pca_plot3d
  - Update Rd files to include pca_plot3d

catterbu commented:
@nsh87 I just pushed what should be the final version of the pca plot stuff. All the rebasing and everything should finally lead to seemless from this point. the boxplot is ready too, but I will need to rebase off **dev** after this is merged before that one can be too.

nsh87 commented:
What changes are in the box plot?

catterbu commented:
it will be a separate pull request shortly. that is the one for handling boolean data.

nsh87 commented:
I don't think you'll need to rebase that off dev if I merge all your stuff first/in the order of your commits.
